modules/rostests/apitests/shell32/SHGetKnownFolderPath.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Test for SHGetKnownFolderPath
+ * COPYRIGHT:   Copyright 2025 Petru Răzvan (petrurazvan@proton.me)
+ */
+#include "shelltest.h"
+
+typedef HRESULT (WINAPI *PSHGETKNOWNFOLDERPATH)(REFKNOWNFOLDERID, DWORD, HANDLE, PWSTR *);
+
+static PSHGETKNOWNFOLDERPATH pSHGetKnownFolderPath = NULL;
+
+START_TEST(SHGetKnownFolderPath)
+{
+    HRESULT hr;
+    HINSTANCE hShell32;
+    PWSTR path = NULL;
+    HRESULT result;
+
+    hShell32 = LoadLibraryW(L"shell32.dll");
+    if (!hShell32)
+    {
+        skip("Failed to load shell32.dll\n");
+        return;
+    }
+
+    pSHGetKnownFolderPath = (PSHGETKNOWNFOLDERPATH)GetProcAddress(hShell32, "SHGetKnownFolderPath");
+
+    if (!pSHGetKnownFolderPath)
+    {
+        trace("SHGetKnownFolderPath not exported. Likely running on an OS older than Vista.\n");
+    }
+
+    hr = CoInitializeEx(NULL, COINIT_APARTMENTTHREADED);
+    if (FAILED(hr))
+    {
+        skip("Failed to initialize COM library. Error Code: 0x%08lX\n", hr);
+        pSHGetKnownFolderPath = NULL;
+    }
+
+    if (pSHGetKnownFolderPath)
+    {
+        result = pSHGetKnownFolderPath(FOLDERID_Desktop, 0, NULL, &path);
+        if (SUCCEEDED(result))
```

Comment:
Are you sure?

https://learn.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetknownfolderpath
'Returns S_OK if successful, or an error value otherwise'